### PR TITLE
tests(btcindexer): verifyConfirmingBlocks and getSenderAddresses

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Summary by Sourcery

Add comprehensive tests for BTC indexer’s confirming blocks verification and sender address extraction in block processing

Tests:
- Add tests for verifyConfirmingBlocks covering valid, reorged, empty, and SPV verification failure scenarios
- Add tests for getSenderAddresses (via processBlock) handling Electrs API success, failure, and invalid responses

### CODE REVIEW

File: packages/btcindexer/src/btcindexer.test.ts [hunk 1]
```diff
@@ -630,3 +630,176 @@ describe("CFStorage.insertBlockInfo (Stale Block Protection)", () => {
...
+				tx329.id,
+				0,
+				block329.hash,
+				block329.height,
+				tx329.suiAddr,
+				tx329.amountSats,
+				"confirming", // Set as confirming status
+				Date.now(),
+				Date.now(),
+				BtcNet.REGTEST,
```
Comment (Rcc999) [inline packages/btcindexer/src/btcindexer.test.ts]:
```suggestion
				0,
				BtcNet.REGTEST,
```

File: packages/btcindexer/src/btcindexer.test.ts [hunk 2]
```diff
@@ -630,3 +630,176 @@ describe("CFStorage.insertBlockInfo (Stale Block Protection)", () => {
...
+		expect(
+			suiClientSpy,
+			"Verify that verifyBlocks was called with the correct block hash",
+		).toHaveBeenCalledWith([block329.hash]);
+
+		// Check that the transaction status remains 'confirming' since block is still valid
+		await verifyMintingStatus("confirming", db, tx329.id);
+	});
+
+	it("should handle empty confirming blocks list", async () => {
```
Comment (Rcc999) [inline packages/btcindexer/src/btcindexer.test.ts]: We assume here that the db is empty but we need to check first if there is no confirming blocks in the db right

Comment (sczembor) [reply packages/btcindexer/src/btcindexer.test.ts]: look at after.each
